code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -384,44 +489,24 @@
 		if(user.loc == src) //so we don't get the message if we resisted multiple times and succeeded.
 			to_chat(user, "<span class='warning'>You fail to break out of [src]!</span>")
 
-/obj/structure/closet/proc/bust_open()
-	welded = FALSE //applies to all lockers
-	locked = FALSE //applies to critter crates and secure lockers only
-	broken = 1 //applies to secure lockers only
-	open()
-
 /obj/structure/closet/AltClick(mob/user)
 	..()
 	if(!user.canUseTopic(src, be_close=TRUE) || !isturf(loc))
 		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
 		return
-	if(opened || !secure)
-		return
-	else
-		togglelock(user)
-
-/obj/structure/closet/proc/togglelock(mob/living/user)
-	if(secure && !broken)
-		if(allowed(user))
-			if(iscarbon(user))
-				add_fingerprint(user)
-			locked = !locked
-			user.visible_message("<span class='notice'>[user] [locked ? null : "un"]locks [src].</span>",
-							"<span class='notice'>You [locked ? null : "un"]lock [src].</span>")
-			update_icon()
-		else
-			to_chat(user, "<span class='notice'>Access Denied</span>")
-	else if(secure && broken)
-		to_chat(user, "<span class='warning'>\The [src] is broken!</span>")
+	togglelock(user)
 
 /obj/structure/closet/emag_act(mob/user)
 	if(secure && !broken)
 		user.visible_message("<span class='warning'>Sparks fly from [src]!</span>",
 						"<span class='warning'>You scramble [src]'s lock, breaking it open!</span>",
 						"<span class='italics'>You hear a faint electrical spark.</span>")
 		playsound(src, "sparks", 50, 1)
-		broken = 1
+		broken = TRUE
 		locked = FALSE
+		if(!QDELETED(lockerelectronics))

```

Comment:
An emag should break the electronics. Not remove it completely from the world. Same behavior as with emagging an airlock.

Reply:
Emagging an airlock makes you discard the installed electronics when you deconstruct it, though. I can change it to just wipe access from the electronics though